charmcraft/charm_builder.py

```diff
@@ -228,33 +232,50 @@ def _install_dependencies(self, staging_venv_dir):
         # create virtualenv using the host environment python
         with instrum.Timer("Creating venv"):
             _process_run(["python3", "-m", "venv", str(staging_venv_dir)])
-        pip_cmd = str(_find_venv_bin(staging_venv_dir, "pip3"))
+        pip_cmd = str(_find_venv_bin(staging_venv_dir, "pip"))
 
-        with instrum.Timer("Getting pip version"):
-            _process_run([pip_cmd, "--version"])
+        with instrum.Timer("Ensuring a recent enough pip..."):
+            if get_pip_version(pip_cmd) < (22, 0):
```

Comment:
Can you add a comment on why `22.0` is the minimum?

Reply:
Strictly speaking I don't know whether that's the actual minimum version needed (it could be lower), but the version of pip included with `focal` has dependency resolution issues. Comment added.